src/jep/InvocationHandler.java

```diff
@@ -119,6 +122,14 @@ public Object invoke(Object proxy, Method method, Object[] args)
             throws Throwable {
         this.jep.isValidThread();
 
+        if (this.functionalInterface && (method.getModifiers() & Modifier.ABSTRACT) == 0) {
+            /*
+             * If it's a non-abstract (default) interface method,
+             * fallback to invoking the default implementation if we're a functional interface.
+             */
+            return method.invoke(proxy, args);
```

Comment:
I've done more investigation into handling default methods. The generally accepted solution is to use MethodHandles, however that is not available until java 1.7. I think we could write some JNI that uses CallNonvirtual<type>Method to accomplish the same thing but it would be an annoyingly large amount of code. It looks like this would make a good excuse to drop support of java 1.6.

Since its getting complicated and default method handling is already somewhat broken anyway I think we should not try to fix it with this pull request. I do think we should try to fail more gracefully than a StackOverflowError, the best option I see is to just use invoke with functionalInterfacec=false, which will probably end up with an Attribute Not Found type of error. Those errors would be slightly easier to identify, and if we don't solve default methods it would also give developers the ability to reimplement the default method in python to avoid errors.